rados/ioctx.go

```diff
@@ -302,22 +303,36 @@ type ObjectListFunc func(oid string)
 // RadosAllNamespaces before calling this function to return objects from all
 // namespaces
 func (ioctx *IOContext) ListObjects(listFn ObjectListFunc) error {
-	var ctx C.rados_list_ctx_t
-	ret := C.rados_nobjects_list_open(ioctx.ioctx, &ctx)
-	if ret < 0 {
-		return getError(ret)
+	page_results := C.size_t(1000)
+	var filter_len C.size_t
+	results := make([]C.rados_object_list_item, page_results)
+
+	next := C.rados_object_list_begin(ioctx.ioctx)
+	if next == nil {
+		return errors.New("Pool does not exist")
+	}
+	defer C.rados_object_list_cursor_free(ioctx.ioctx, next)
+	finish := C.rados_object_list_end(ioctx.ioctx)
+	if finish == nil {
+		return errors.New("Pool does not exist")
 	}
-	defer func() { C.rados_nobjects_list_close(ctx) }()
+	defer C.rados_object_list_cursor_free(ioctx.ioctx, finish)
 
 	for {
-		var cEntry *C.char
-		ret := C.rados_nobjects_list_next(ctx, &cEntry, nil, nil)
-		if ret == -C.ENOENT {
-			return nil
-		} else if ret < 0 {
+		ret := C.rados_object_list(ioctx.ioctx, next, finish, page_results, nil, filter_len, (*C.rados_object_list_item)(unsafe.Pointer(&results[0])), &next)
+		if ret < 0 {
 			return getError(ret)
 		}
-		listFn(C.GoString(cEntry))
+
+		numEntries := int(ret)
+		for i := 0; i < numEntries; i++ {
+			item := results[i]
+			listFn(C.GoStringN(item.oid, (C.int)(item.oid_length)))
+		}
+
+		if ret := C.rados_object_list_is_end(ioctx.ioctx, next); int(ret) == 1 {
```

Comment:
Again, it would make things clearer to have the "magic" value 1 as a real constant such as `const listEndSentinel = C.int(1)`. Also, since ret isn't used elsewhere again there's no need to assign ret, just do:
```
if C.rados_object_list_is_end(...) == listEndSentinel {
...
}
```